src/Theme/index.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { StyleSheet } from "react-native";
+import defaultBranding from "../branding_defaults";
+
+export const ThemeContext = React.createContext({
+  theme: {}
+});
+
+export class Provider extends React.Component {
+  themeConfigs = {};
+  branding = defaultBranding;
+
+  constructor(props) {
+    super(props);
+    if (props.branding) {
+      this.branding = deepMerge(this.branding, props.branding);
+    }
+  }
+
+  updateThemeConfig = (name, themeConfig) => {
+    if (this.props.theme && this.props.theme[name])
+      themeConfig = mergeDeep(themeConfig, this.props.theme[name]);
+
+    themeConfig = JSON.parse(
+      JSON.stringify(themeConfig).replace(/@([\w_-]+)/gm, (match, key) => {
+        if (!this.branding[key]) console.log(key);
+
+        return this.branding[key];
+      })
+    );
+
+    return (this.themeConfigs[name] = {
+      computedStyle: themeConfig.style
+        ? StyleSheet.create(themeConfig.style)
+        : {},
+      props: themeConfig.props
+    });
+  };
+
+  render() {
+    return (
+      <ThemeContext.Provider
+        value={{
+          theme: this.themeConfigs,
+          updateThemeConfig: this.updateThemeConfig
+        }}
+      >
+        {this.props.children}
+      </ThemeContext.Provider>
+    );
+  }
+}
+
+export function withTheme(componentName, ThemedComponent) {
+  const propTypes = {
+    ...ThemedComponent.propTypes
+  };
+  delete propTypes.theme;
+
+  return class extends React.Component {
+    static displayName = componentName;
+    static propTypes = {
+      ...propTypes
+    };
+    static defaultProps = {
+      ...ThemedComponent.defaultProps
+    };
+    constructor() {
+      super();
+      try {
+        ThemeContext._currentValue.updateThemeConfig(
+          componentName,
+          ThemedComponent.themeConfig
+        );
+      } catch (e) {
+        console.warn(`failed to update theme for ${componentName}`, e);
+      }
+    }
+
+    render() {
+      return (
+        <ThemeContext.Consumer>
+          {({ theme }) => {
+            const computedTheme =
+              (theme[componentName] || {}).computedStyle || {};
+
+            const fullTheme = this.props.theme
+              ? mergeDeep({}, computedTheme, this.props.theme)
+              : computedTheme;
+
+            return (
+              <ThemedComponent
+                {...(theme[componentName] || {}).props || {}}
+                {...this.props}
+                theme={fullTheme}
+              />
+            );
+          }}
+        </ThemeContext.Consumer>
+      );
+    }
+  };
+}
+
+function isObject(item) {
```

Comment:
Should not we test those utils? Or use specific dependencies like https://github.com/sindresorhus/is-obj (basically everything is made by sindresorhus 😀)